src/components/PeoplePage.tsx

```diff
@@ -1,8 +1,120 @@
+import { useEffect, useMemo, useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+import { Person } from '../types';
+import { getPeople } from '../api';
+
+interface FilterParams {
+  sex: string | null,
+  centuries: string[] | null,
+  input: string | null,
+  sort: keyof Person | null,
+  order: string | null,
+}
+
+type Prepare = (
+  people: Person[],
+  filterParams: FilterParams,
+) => Person[];
+
+const preparePeople: Prepare = (people, {
+  sex,
+  centuries,
+  input,
+  sort,
+  order,
+}) => {
+  return people
+    .map((person) => {
+      return {
+        ...person,
+        motherName: person.motherName || '-',
+        fatherName: person.fatherName || '-',
+        mother: people.find(mother => mother.name === person.motherName),
+        father: people.find(father => father.name === person.fatherName),
+      };
+    })
+
+    .filter(person => {
+      const isSexValid = sex === null ? true : person.sex === sex;
+
+      const isCenturyValid = centuries === null || !centuries?.length
+        ? true
+        : centuries.includes(Math.ceil(person.born / 100).toString());
+
+      const isNameValid = input === null
+        ? true
+        : person.name.toLowerCase().includes(input.toLowerCase());
```

Comment:
Didn't get logig: If vlue is equal to null - it's valid value?
```suggestion
      const isSexEqual = person?.sex === sex;

      const isCenturyIncludes = centuries?.includes(Math.ceil(person.born / 100).toString());

      const isNameEqual = person.name.toLowerCase().includes(input?.toLowerCase() || '');
```